src/ppx/grfx/grfx_swapchain.cpp

```diff
@@ -143,6 +129,67 @@ void Swapchain::Destroy()
     grfx::DeviceObject<grfx::SwapchainCreateInfo>::Destroy();
 }
 
+Result Swapchain::CreateColorImages()
+{
+    if (!mColorImages.empty()) {
+        return ppx::SUCCESS;
+    }
+    for (uint32_t i = 0; i < mCreateInfo.imageCount; ++i) {
+        grfx::ImageCreateInfo rtCreateInfo = ImageCreateInfo::RenderTarget2D(mCreateInfo.width, mCreateInfo.height, mCreateInfo.colorFormat);
+        rtCreateInfo.ownership             = grfx::OWNERSHIP_RESTRICTED;
+        rtCreateInfo.RTVClearValue         = {0.0f, 0.0f, 0.0f, 0.0f};
+        rtCreateInfo.initialState          = grfx::RESOURCE_STATE_PRESENT;
+        rtCreateInfo.usageFlags =
```

Comment:
do we need IMAGE_USAGE_STORAGE to be consistent with CreateColorImagesFromApiObjects?

Reply:
We can probably remove storage image usage flag.
`CreateColorImages` and `CreateColorImagesFromApiObjects` could share more of their parameters, however I'm not going to try to rewrite anything here (this change is mostly moving things around).